framework/encode/vulkan_state_writer.cpp

```diff
@@ -408,7 +408,9 @@ void VulkanStateWriter::WriteSemaphoreState(const VulkanStateTable& state_table)
             }
 
             // Write command to set semaphore value on replay
-            WriteSignalSemaphoreValue(signal_call_id, device_wrapper->handle_id, wrapper->handle_id, semaphore_value);
+            if (semaphore_value)
```

Comment:
Since `semaphore_value` is an increasing `uint64_t` value for `VK_SEMAPHORE_TYPE_TIMELINE`, as you note, would you please make this an explicit comparison against `!= 0`?  (We've been tending towards explicit comparisons for clarity.)